src/Spice86.Core/Emulator/Devices/Video/AeonCard.cs

```diff
@@ -0,0 +1,836 @@
+                AttributeController.WriteRegister(_attributeRegister, value);
+                break;
+
+            case Ports.CrtControllerAddress:
+            case Ports.CrtControllerAddressAlt:
+                _crtRegister = (CrtControllerRegister)value;
+                break;
+
+            case Ports.CrtControllerData:
+            case Ports.CrtControllerDataAlt:
+                int previousVerticalEnd = CrtController.VerticalDisplayEnd;
+                CrtController.WriteRegister(_crtRegister, value);
+                if (previousVerticalEnd != CrtController.VerticalDisplayEnd) {
+                    ChangeVerticalEnd();
+                }
+
+                break;
+        }
+    }
+
+    /// <summary>
+    /// Special shortcut for VGA controller to select a register and write a value in one call.
+    /// </summary>
+    public void WriteWord(int port, ushort value) {
+        WriteByte(port, (byte)(value & 0xFF));
+        WriteByte(port + 1, (byte)(value >> 8));
+    }
+
+    public void WriteDWord(int port, uint value) {
+        throw new NotImplementedException("WriteDWord() is not implemented for VGA controller.");
+    }
+
+    /// <summary>
+    ///     Gets the VGA DAC.
+    /// </summary>
+    public Dac Dac { get; } = new();
+
+    /// <summary>
+    ///     Gets the VGA graphics controller.
+    /// </summary>
+    public Graphics Graphics { get; } = new();
+
+    /// <summary>
+    ///     Gets the VGA sequencer.
+    /// </summary>
+    public Sequencer Sequencer { get; } = new();
+
+    /// <summary>
+    ///     Gets the VGA CRT controller.
+    /// </summary>
+    public CrtController CrtController { get; } = new();
+
+    /// <summary>
+    ///     Gets the current display mode.
+    /// </summary>
+    public VideoMode CurrentMode { get; private set; } = null!;
+
+    /// <summary>
+    ///     Gets the VGA attribute controller.
+    /// </summary>
+    public AttributeController AttributeController { get; } = new();
+
+    /// <summary>
+    ///     Gets a pointer to the emulated video RAM.
+    /// </summary>
+    public nint VideoRam { get; }
+
+    /// <summary>
+    ///     Gets the text-mode display instance.
+    /// </summary>
+    public TextConsole TextConsole { get; }
+
+    public void SetVramByte(uint address, byte value) {
+        CurrentMode.SetVramByte(address, value);
+    }
+
+    public void Render(uint address, object width, object height, nint pixelsAddress) {
+        _presenter ??= GetPresenter();
+        _presenter.Update(pixelsAddress);
+    }
+
+    public void TickRetrace() {
+        // Inactive at tick time, but will become active once the code checks for it.
+        _crtStatusRegister = StatusRegisterRetraceInactive;
+    }
+
+    public void UpdateScreen() {
+        _gui?.UpdateScreen();
+    }
+
+    public event EventHandler? VideoModeChanged;
+
+    private void FillDispatchTable() {
+        _dispatchTable.Add(0x00, new Callback.Callback(0x00, SetVideoMode));
```

Comment:
Remark: for rewritten code sake, would be nice if all the functions being called like this were public, so that we can write SetCursorType();
instead of
MOV AX,00
INT 10h
:)

Reply:
Done